src/Processors/QueryPlan/Optimizations/optimizePrewhere.cpp

```diff
@@ -45,6 +46,10 @@ void optimizePrewhere(Stack & stack, QueryPlan::Nodes &)
     if (!source_step_with_filter)
         return;
 
+    /// Prewhere is broken with StorageMerge and old analyzer.
+    if (!source_step_with_filter->getContext()->getSettingsRef().allow_experimental_analyzer && typeid_cast<ReadFromMerge *>(source_step_with_filter))
```

Comment:
For example this one I've been investigated is related https://github.com/ClickHouse/ClickHouse/issues/63105 
But what are other issues?

Reply:
That issue I'm trying to fix by recalculating child plans after prewhere info is updated https://github.com/ClickHouse/ClickHouse/pull/64893